pennylane/gradients/pulse_generator_gradient.py

```diff
@@ -702,3 +709,15 @@ def expand_invalid_trainable_pulse_generator(x, *args, **kwargs):
 pulse_generator = gradient_transform(
     _pulse_generator, expand_fn=expand_invalid_trainable_pulse_generator
 )
+
+
+@pulse_generator.custom_qnode_wrapper
+def pulse_generator_qnode_wrapper(self, qnode, targs, tkwargs):
+    """A custom QNode wrapper for the gradient transform :func:`~.pulse_generator`.
+    It is equivalent to the default QNode wrapper of any :func:`~.gradient_transform`,
+    but in addition warns that applying ``pulse_generator`` to a ``QNode`` directly
+    is only supported for scalar pulse parameters.
+    """
+    transform_name = "pulse generator parameter-shift"
+    warn_pulse_diff_on_qnode(transform_name)
+    return self.default_qnode_wrapper(qnode, targs, tkwargs)
```

Comment:
Will this warning come up when the transform is used as part of the JAX entrypoints like `jax.grad`?

Reply:
No, because these entrypoints use the tape transform. (That's also why they actually support non-scalar parameters already :))